### DISCUSSION TITLE
Highlight the main channel

### DISCUSSION THREAD

**Discussion by ShakhmatovIS** [PMEAL/porespy#900] [General]
Hello to all researchers, colleagues, scientists.
I have a small problem.
A sample after acid injection and I need to isolate 1 main channel that connects the inlet and outlet of the sample. I need to exclude all caverns not participating in filtration and get the characteristics of only the main cavern.
Is there any functionality in porespy or openpnm to solve this problem?
![222](https://github.com/PMEAL/porespy/assets/13432456/de9670c8-445b-419e-b7bb-d41b7a9e214e)
Thanks so much in advance for your help

**Comment by jgostick** [comment]
There are probably "official" graph theory ways to do this, but I would say just do a diffusion simulation and find all the 'plateaus' in the concentration profiles...these will be your dead ends.  Since no flow goes through the dead ends, there will be no concentration gradient, so this is the signature you're looking for.  The only issue is that you need to numerically define a plateau.  Like two sites whose relative or absolute concentration is 1e-6 are on a plateau?  This works on the pore network as well as the image if your image is not too large to compute the diffusivity using something like TauFactor.

**Comment by ShakhmatovIS** [comment]
Thanks professor, I will try to make such calculation through filtration modeling. I hope it works.
